services/auth/src/api.js

```diff
@@ -844,7 +844,7 @@ builder.declare({
 
       // Check scopes
       const formerRoleScopes = this.resolver.resolve(role.scopes);
-      const scopesAdded = input.scopes.filter(s => !scopeUtils.scopeMatch(formerRoleScopes, [[s]]));
+      const scopesAdded = input.scopes.filter(s => !scopeUtils.satisfiesExpression(formerRoleScopes, s));
```

Comment:
just to make sure: so we are now sure to expect the new syntax..... is it an array? Not an object like `{AllOf: []}` or `{AnyOf: []}` etc?

Reply:
In this case, `s` is a single scope (a string).

Reply:
sorry, I meant `input.scopes`

Reply:
Basically, I just wanted to double check that we don't loop with `Array.prototype.filter` over an object. If this worry is unfounded, r+

Reply:
Good question -- yeah, that's enforced with JSON-schema.